### ISSUE TITLE
SpinnerList: changing item renderer dimensions does not update scroll position

### ISSUE THREAD

**Issue by Adrian-S** [BowlerHatLLC/feathers#1493]
Hey, we are trying to make some changes to the SpinnerList:
- items should be aligned to the left side, not centred
- have a strict number of columns visible at all times, we want 2 or 3 columns visible, not 2.5, we are changing the layout based on iPad orientation.

We almost got this with minimal changes:
1. In `HorizontalSpinnerLayout` we set: `var horizontalAlignOffsetX:Number = 0;`

This moved the centred item to the left and seams to work fine. Should we account for anything else?

2. Rotating the device, causes the spinner to not align the first item on the left edge. Setting the scrollToPageIndex, fixes the problem but only if the pageIndex is above 7 or below -8 (this was with 10 items in dataProvider and with `horizontalAlignOffsetX = 0`).

The attached file will replicate the problem, and you can see them with an offset, not centred. This example is using the default item renderer and it's width is set to 1/2 or 1/3 of the screen on an iPad (label stays on the left of the item). You can test with or without setting the `horizontalAlignOffsetX = 0`, problem is still there.

**Basically the items don't get properly aligned** from portrait to landscape, and when trying to left align the items the error is obvious.

[ListScreenTest.as.txt](https://github.com/BowlerHatLLC/feathers/files/591962/ListScreenTest.as.txt)

**Comment by raresn** [comment]
I have exactly the same issue.

**Comment by joshtynjala** [comment]
I haven't had a chance to look at this yet, but I will as soon as I can.

**Comment by raresn** [comment]
Thank you Josh! Much appreciated!

**Comment by joshtynjala** [comment]
Okay, so it looks like there's a bug in SpinnerList. SpinnerList does not adjust the scroll position correctly after the dimensions of the item renderers change. scrollToPageIndex() doesn't help as a workaround because it incorrectly assumes that you're already on the correct page.

**Comment by Adrian-S** [comment]
Josh, one more thing I've noticed, that produces unexpected results:
```
spinnerLayout.typicalItemHeight = 300;
spinner.height = 300;
```
For a `typicalItemHeight` higher than `spinner.height - 2`, left / right scrolling will jump by 2 or 3 items.
I was not able to find why the `-2`, but I know it's related to `ListDataViewPort` used, that is somehow at `y = 1`, and I assume it's height is missing 2.

PS: I'm doing a calendar app, and I'm using the spinner list to make an infinite scroll for days. Since it's possible to scroll to a negative index, this helps me because I can go any number of days left / right. I've already added a workaround to send the real index to the item renderer via the dataProvider.

**Comment by joshtynjala** [comment]
Does the SpinnerList have padding (not its layout, but the list)? That would explain why the view port is not at y=0. If there's padding, you would want to set the typicalItemHeight smaller than the list's height to account for that.

**Comment by Adrian-S** [comment]
Yes you were right, padding top and bottom was added from MetalWorksMobileTheme.

But still `typicalItemHeight` higher than `spinner.height`, will make it jump over multiple items, I think the `horizontalPageIndex` is calculated with the entire width of the spinner, not the spinner items. This is somewhat strange, since height should play no role on a HorizontalSpinnerLayout.

Also one other thing that bugs me about `horizontalPageIndex`, is that is not calculated based on the middle of the item. Scrolling the spinner a few pixels to the right will change the index, while scrolling to the left will only change the index after the entire item has been moved.

**Comment by dannycortesv** [comment]
I'm still experience this on Android, iOS is working well, but select the last item of the spinnerList is not possible on Android, look: https://www.screencast.com/t/OuBiLCqYnQD

**Comment by joshtynjala** [comment]
@dannycortesv I assume that you're using the latest code from Github? If yes, then please put together some simple code that will consistently reproduce the issue on Android. Thanks!

**Comment by dannycortesv** [comment]
Yes, sure, this is:

`
spinner = new SpinnerList();
SpinnerStyle.setFormat(spinner);
this.addChild(spinner);
spinner.x=Main.sW*0.32;
spinner.width=Main.sW*0.53;

public static function setFormat(spinner:SpinnerList,textureName:String="grid6"):void{
			spinner.height=int(Main.sH*0.1);
			spinner.x=Main.sW*0.1;
			
			spinner.backgroundSkin = Grids.getAlternativeGrid(textureName);
			spinner.width=int(Main.sW*0.8);	
			spinner.height=int(Main.sH*0.1);
			
			spinner.itemRendererFactory = function():DefaultListItemRenderer {
				var renderer:DateSpinnerListItemRenderer = new DateSpinnerListItemRenderer();
				renderer.defaultLabelProperties.textFormat = new BitmapFontTextFormat(Common.fontName,int(Common.fontSizeNormal*0.85));
				return renderer;
			};
			spinner.horizontalScrollPolicy=ScrollPolicy.OFF;
		}


package tetheree.renderers
{
	import feathers.controls.renderers.DefaultListItemRenderer;

	public class DateSpinnerListItemRenderer extends DefaultListItemRenderer
	{
		override protected function draw():void {
			super.draw();
			labelTextRenderer.alpha = isSelected ? 1 : 0.5;
		}
	}
}
`

**Comment by dannycortesv** [comment]
Created #1532 for this, i think it's different than this.
